test/test_cuda.py

```diff
@@ -248,45 +248,62 @@ def tmp(t):
 # - disable inplace test, if set to True, no inplace test will be done (default=False)
 # - decorator, e.g., unittest.skipIf (default is no decorator)
 tests = [
-    ('add', small_3d, lambda t: [number(3.14, 3, t)]),
+    ('add', small_3d, lambda t: [number(3.14, 3, t)], '', types, False,
+        "skipIfRocm:ByteTensor,CharTensor,HalfTensor,ShortTensor"),
```

Comment:
Spelling all of this out is a bit weird. Do you really need to specify for each test, individually, which data types work, or are there well defined "classes" of tensors which do or don't work? I'm seeing something like Half + Small Integral (Byte,Char,Short) + Big Integral (Int,Long) + (Float,Double)